bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
+  #If they do then it returns the set path
+  def statefile
+   setting_exists?('statefile')
+   if file_exist?('statefile') == false
+    #Create the necessary configuration within state.yaml
+    #This only runs if the file was just created within file_exist?
+    open(@config['statefile'], 'a') { |f|
+     f << "---\n"
+     f << "job1:\n"
+     f << "  :forward: true\n"
+     f << "  :rewind: false\n"
+     f << "job2:\n"
+     f << "  :forward: true\n"
+     f << "  :rewind: false\n" }
+   end
+   @config['statefile']
+  end
  end
- 
- #Saves the current state to file
- def save!
-  File.open(@statefile, 'w') { |f| YAML.dump(@sf, f) }
+  
+ #Returns/Creates the config object
+ def self.config
+  @config ||= Config::new 
  end
 
- #Checks the current state of the requested script for the given job
- def status(name, command, event)
-  #The script will run if it's state is currently false
-  if @sf[name][:"#{command}"] == false
-  else
-   event.log.error "Can't execute #{command} for #{name} as it is already set to true"
-   exit 1 
-  end
+ def config
+  ConfigFile::config
  end
 end
 
-class Config
- def initialize(config)
-  @configfile = config
-  @config = YAML.load_file(@configfile) 
- end
- 
- #Sets the directory for any given setting in the file 
- def dirset(path, name, setting)
-  if exists?(setting)
-   @dir = @config[setting]
-   
-   #If the setting exists it checks if it is absolute or relative
-   if @dir.include? "../"
-    @dir = File.join(path, @dir, name)
-   else
-    @dir = File.join(@dir, name)
-   end
-  else
-   puts "The config file does not contain a path for the setting: #{setting}"
-   if setting == 'jobdir' then exit 1 end
+#Logger Module
+module EventLogger
+ class Log
+  include ConfigFile
+
+  #Configure the logger
+  def initialize
+    @logger = Logger.new(config.logfile)
+    @logger.level = Logger::DEBUG
+    @logger.formatter = proc do |severity, datetime, progname, msg|
+     date_format = datetime.strftime("%d-%m-%Y %H:%M:%S")
+     "[#{date_format}]  #{severity.ljust(5)}: #{msg}\n"
+    end
+    
+    #Redirect STDERR messages and append to the log file
+    $stderr.reopen(config.logfile, "a")
+  end
+  
+  #This does something
+  def method_missing(m,*args,&block)
+    log.send(m,*args,&block)
+  end
+  
+  #Return the logger
+  def log
+   @logger
   end
  end
  
- #Check to see if a given setting exists in the file
- def exists?(setting)
-  if @config[setting] != nil then return true end
+ #Returns/Creates the logger
+ def self.log
+   @log ||= Log::new
  end
+
+ def log
+  EventLogger::log
+ end 
 end
 
 class Job
- def initialize(jobdir, cmdscript, name, command, log)
-  @jobdir = jobdir
-  @cmdscript = cmdscript
-  @job = name
-  @command = command
-  @event = log
- end
+ class State
+  include EventLogger
+  include ConfigFile
+
+  #Toggles value in state hash
+  def toggle(name) 
+   @sf = YAML.load_file(config.statefile)
+
+   #Finds all true/false values in hash
+   t = @sf[name].key(true)
```

Comment:
I'm not sure why this is happening like this, we just need to work with the  forward/rewind keys. I'd consider doing forward and rewind methods on this class instead of toggle. 
```
def forward
  set forward to true, rewind to false
end

def rewind
  set rewind to true, forward to false
end
```

Reply:
The toggling now functions as two separate methods; forward and rewind